smack-omemo/src/main/java/org/jivesoftware/smackx/omemo/element/OmemoBundleElement.java

```diff
@@ -76,18 +96,33 @@ public OmemoBundleElement(int signedPreKeyId, String signedPreKeyB64, String sig
      * @param preKeys HashMap of preKeys
      */
     public OmemoBundleElement(int signedPreKeyId, byte[] signedPreKey, byte[] signedPreKeySig, byte[] identityKey, HashMap<Integer, byte[]> preKeys) {
+        if (signedPreKeyId <= 0) {
```

Comment:
That is exactly the kind of code duplication that you want to avoid. The usual solution is to change this constructor to a static constructor method: `public static OmemoBundleElement newOmemoBundleElement(int signedPreKeyId, byte[] signedPreKey, byte[] signedPreKeySig, byte[] identityKey, HashMap<Integer, byte[]> preKeys)` which internally used the constructor above.